### PR TITLE
Overlay documentation

### PR DESCRIPTION
**Pull request by danielwilms**
The documentation summarises idea and useage of overlays,
as implemented in https://github.com/COVESA/vss-tools/pull/173

Detailed documentation of the CLI will be done in tools.
This just serves as an overview and background.

Signed-off-by: Daniel Wilms <Daniel.DW.Wilms@bmw.de>

### CODE REVIEW

File: docs-gen/content/rule_set/overlay.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+VSS defines the standard catalog for vehicle signals independent of the used protocol and environment.
+In order to adopt the specification we realize that certain additions and modifications to the standard catalog are necessary.
+**VSS Overlay** are meant to bring you a standardized way of handling those changes.
+
+The following features with the intended usage patterns are currently supported:
+1. **Adding new nodes:** By adding nodes the standard catalog can be extended with proprietary concepts.
+1. **Changing the value of existing metadata:** The standard catalog defines metadata based on what is assumed to be an average vehicle. 
+Configurations may differ slightly (e.g. the instantiation of number of available seats), or special situations that require a (limited) modification of existing metadata.
+1. **Adding new key/value pairs as additional metadata:** 
+Extending the model is with richer information is a fundamental feature enabled by the layer concept.  For example, deploying VSS into a specific scenario or with a particular binding/technology often needs some additional information.
```
Comment (erikbosch) [inline docs-gen/content/rule_set/overlay.md:16]: Remove `is` in `model is with`.

File: docs-gen/content/rule_set/overlay.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+1. **Multiple layer files:** VSS layers can be split into several files in order to clearly separate concerns. Layering allows all the features above to be applied in a composable manner. In order to keep a determinstic result a clear order has to remain.
+
+### Scenarios for using Overlays
+
+The standard catalogue is specified within COVESA and defines a common view of the major
+attributes, signals and actuators of vehicles. This is used in many different scenarios,
```
Comment (erikbosch) [inline docs-gen/content/rule_set/overlay.md:22]: Shall be `attributes, sensors and actuators` if we keep the view that everything is signals.

**Comment by erikbosch** [comment]
For `overlay.drawio.png` - what is the source format? Would it make sense to include the source format as well (or link to where it is if public) so the image can be updated if needed?

File: docs-gen/content/rule_set/private_branch.md [hunk 3]
```diff
@@ -1,54 +0,0 @@
----
-title: "Private Branch"
```
Comment (erikbosch) [inline docs-gen/content/rule_set/private_branch.md:2]: What is our view on the Private branch. With the overlay functionality OEMs can easily add data to existing branches, but do we actually see the private branch as obsolete, or do we still see `Private` as a reasonable naming if you are to add signals that currently do not have any "logic" home in VSS. I can see two options and I have not yet made up my mind:

- Either we see "private" as  a good naming convention for extensions, and then we can briefly mention that in the "overlay.md" document, e.g. "If you are to add OEM-specific signals that does not fit into the VSS structure, consider adding them to Vehicle.Private.Something.
- Or we see "private" as obsolete, but then we should preferably remove "private" from the existing vspec files.

**Comment by danielwilms** [comment]
> For `overlay.drawio.png` - what is the source format? Would it make sense to include the source format as well (or link to where it is if public) so the image can be updated if needed?

It's drawn in draw.io and a copy of the diagram is encoded in the png. you can open it [here in the editor](https://app.diagrams.net/#Uhttps%3A%2F%2Fgithub.com%2Fdanielwilms%2Fvehicle_signal_specification%2Fraw%2F2022-06-28-overlay%2Fdocs-gen%2Fstatic%2Fimages%2Foverlay.drawio.png)

File: docs-gen/content/rule_set/private_branch.md (hunk 3 above)
Comment (danielwilms) [reply docs-gen/content/rule_set/private_branch.md:2]: I would see it as obsolete. I wanted to delete as well the folder in the spec file, but forgot. I described it as one scenario for overlay. But let's discuss it in the call later.

File: docs-gen/content/rule_set/overlay.md (hunk 1 above)
Comment (danielwilms) [reply docs-gen/content/rule_set/overlay.md]: done

File: docs-gen/content/rule_set/overlay.md (hunk 2 above)
Comment (danielwilms) [reply docs-gen/content/rule_set/overlay.md]: done.

File: docs-gen/content/rule_set/private_branch.md (hunk 3 above)
Comment (danielwilms) [reply docs-gen/content/rule_set/private_branch.md:2]: General opinion leads towards removing private as obsolete:
- @danielwilms create issue for tools to remove old "private" code.
- @danielwilms remove empty private folder

**Comment by SebastianSchildt** [comment]
If no further comments until July 12th, this can be merged

File: docs-gen/content/rule_set/overlay.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
+---
+title: "Overlay"
+date: 2019-08-04T12:46:30+02:00
+weight: 7
+---
+
+VSS defines the standard catalog for vehicle signals independent of the used protocol and environment.
+In order to adopt the specification we realize that certain additions and modifications to the standard catalog are necessary.
+**VSS Overlay** are meant to bring you a standardized way of handling those changes.
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:9]: VSS overlayS  are 

or

VSS Overlay IS

File: docs-gen/content/rule_set/overlay.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+The following features with the intended usage patterns are currently supported:
+1. **Adding new nodes:** By adding nodes the standard catalog can be extended with proprietary concepts.
+1. **Changing the value of existing metadata:** The standard catalog defines metadata based on what is assumed to be an average vehicle. 
+Configurations may differ slightly (e.g. the instantiation of number of available seats), or special situations that require a (limited) modification of existing metadata.
+1. **Adding new key/value pairs as additional metadata:**
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:15]: That is true, but only after https://github.com/COVESA/vss-tools/pull/173 is merged

File: docs-gen/content/rule_set/overlay.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+1. **Adding new key/value pairs as additional metadata:** 
+Extending the model with richer information is a fundamental feature enabled by the layer concept.  For example, deploying VSS into a specific scenario or with a particular binding/technology often needs some additional information.
+1. **Multiple layer files:** VSS layers can be split into several files in order to clearly separate concerns. Layering allows all the features above to be applied in a composable manner. In order to keep a determinstic result a clear order has to remain.
+
+### Scenarios for using Overlays
+
+The standard catalogue is specified within COVESA and defines a common view of the major
+attributes, sensors and actuators of vehicles. This is used in many different scenarios,
+protocols and environments. Additional meta data might be required for data-governance,
+-quality or -sources. The instanciation of branches might not fit your vehicle.
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:24]: typo: instanciation -> instantiation

File: docs-gen/content/rule_set/overlay.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+Proprietrary signals might be needed for extending the specification for your own use-cases.
+Up to now you were on your own in these scenarios. Overlays give you now the chance to change
+and modify the standard catalog in a standardized way.
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:27]: remove "now"

File: docs-gen/content/rule_set/overlay.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+### How does it work?
+
+Simply said, the tooling takes next to the original specification file, *n* additional
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:31]: Simply said, the tooling accepts, *n* additional
 spec files, next to the original specification file,  which can overwrite or extend data in the VSS tree defined by the original specification.

File: docs-gen/content/rule_set/overlay.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+spec files, which overwrite the original information, they contain. Before you start
+you should know:
+- **Overlay-files have to be by themselves valid specification files.**
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:34]: ... have to be valid specification files by themselves
(or in their own right?)

File: docs-gen/content/rule_set/overlay.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+*File: overlay_2.vspec*
+
+### Overlays in the standard catalog
+
+With the feature of overlay, we introduced a new folder in the
+repository called `overlays`. In there you'll find two additional folders:
+
+* `Profiles`: Larger overlays to adapt VSS to a specific vehicle category, like motorbikes.
```
Comment (SebastianSchildt) [inline docs-gen/content/rule_set/overlay.md:111]: Should we have those as lower case? (profiles/extensions)

**Review by SebastianSchildt — CHANGES_REQUESTED** [review]
Nice, some smallcommens nonetheless.

Cool picture!

**Pull request merged**